doc/nrf/releases_and_maturity/releases/release-notes-3.1.0.rst

```diff
@@ -0,0 +1,1183 @@
+      This is done to ensure that the rules are properly applied for an nRF Desktop device connected directly over Bluetooth LE.
+      The new udev rules are applied to any HID device that uses the Nordic Semiconductor Vendor ID (regardless of Product ID).
+    * The HID device discovery to ensure that a discovery failure of a HID device would not affect other HID devices.
+      Without this change, problems with discovery of a HID device could lead to skipping discovery and listing of other HID devices (even if the devices work properly).
+
+Integrations
+============
+
+This section provides detailed lists of changes by :ref:`integration <integrations>`.
+
+Google Fast Pair integration
+----------------------------
+
+* Added the :ref:`ug_bt_fast_pair_adv_manager` page that describes how to integrate the :ref:`bt_fast_pair_adv_manager_readme` module in your application.
+* Updated the :ref:`ug_bt_fast_pair` page to mention the availability of the guide for :ref:`ug_bt_fast_pair_adv_manager` that covers the associated helper module.
+  Mentioned applicability of the :ref:`bt_fast_pair_adv_manager_readme` module in the :ref:`ug_bt_fast_pair_advertising` and the :ref:`ug_bt_fast_pair_use_case_locator_tag` sections.
+
+MCUboot
+=======
+
+The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``81315483fcbdf1f1524c2b34a1fd4de6c77cd0f4``, with some |NCS| specific additions.
+
+The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.
+
+The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:
+
+
+* Added:
+
+  * Support as a first-stage immutable bootloader using hardware cryptographic acceleration on nRF54L10 and nRF54L05.
+  * Support for AES-256 image encryption.
+  * Support for hardware downgrade protection.
+  * Support for SHA512 signature verification.
+  * Linking Time Optimization (LTO) support.
+
+* Updated the following security features:
+
+  * HW downgrade prevention.
+  * RAM memory cleanup controlled by the :kconfig:option:`CONFIG_MCUBOOT_CLEANUP_RAM` Kconfig option.
+  * ECIES-X25519 encryption improvements.
+
+* Fixed:
+
+  * Several stability and functionality improvements.
+  * An issue related to referencing the ARM Vector table of the application, which causes jumping to wrong address instead of the application reset vector for some builds when Zephyr LTO (Link Time Optimization) was enabled.
+
+Zephyr
+======
+
+.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.
+
+The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``9a6f116a6aa9b70b517a420247cd8d33bbbbaaa3``, with some |NCS| specific additions.
```

Comment:
Zephyr SHA is incorrect, `0fe59bf1e4b96122c3467295b09a034e399c5ee6` is the latest merge base:
```suggestion
The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``0fe59bf1e4b96122c3467295b09a034e399c5ee6``, with some |NCS| specific additions.
```